text/0000-icons.md

```diff
@@ -65,19 +68,43 @@ A variable containing an inline SVG element that can be placed directly into you
 
 >**This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with a TVA library to understand, and for somebody familiar with the implementation to implement. This should get into specifics and corner-cases, and include examples of how the feature is used. Any new terminology should be defined here.**
 
-SVG icons provided by designers are programmatically optimized and the syntax is modified based on the use case (such as removing `xmlns` attributes for inline).
+SVG icons provided by designers are placed in the `src/svg` folder, then programmatically optimized using `svgo` and saved to `build/svg`.
+
+To allow for control of the color via CSS by inheriting the foreground color, `currentColor` is assigned to the `fill` and/or `stroke` colors by a build script as appropriate (this assumes single-color icons).
+
+Icon files are then converted to framework-specific syntax.
+
+React is our default, which converts svg to `_jsx` syntax using `svg-to-jsx` and `babel`. React icons can then be imported from `@pluralsight/icons` - since it is the default.
+
+Each framework will have its own corresponding folder. E.g., , `@pluralsight/icons/svelte` for Svelte, etc.
+
+Any additional properties (including styling) can be provided by the `headless-styles` package.
+
+The optimized SVG icons in `build/svg` will also be made available using the `files` attribute of package.json to allow usage by any consumer we don't support.
+
+Since the provided variable represents the icon imagery, and is used as an inline svg (or img) element, we can safely change the underlying implementation without affecting the API.
+
+## Tree shaking
+
+To support tree shaking, each icon is transpiled to its own file.
 
-To allow for flexible control over color, `currentColor` is assigned to the fill and/or stroke colors as appropriate (assuming single-color icons. All others can be exempted from this step).
+The index file then individually imports all of the transpiled icons, and exports them collectively.
 
-Icon files are then converted to a more framework-specific syntax (jsx by default) with one additional folder to specify the framework (e.g., `@pluralsight/icons` for React jsx - since it is the default, `@pluralsight/icons/svelte` for Svelte, etc.).
+```js
+import arrowLeft from 'build/react/arrowLeft'
```

Comment:
We should instead use the syntax
```
export { arrowLeft } from 'build/react/arrowLeft'
```
to reduce the file size since we are dealing with over a hundred icons.

Reply:
Ah, ok, forgot we could do that.  I'll make the update.